### DISCUSSION TITLE
How to execute the Junit-5 tests based on dynamic dependency test conditions

### DISCUSSION THREAD

**Discussion by SCSVEL** [junit-team/junit-framework#2906] [Q&A] [answered]
I am trying to run the tests based on dependency satisfaction like below, and tried something mentioned here but didn't help.
Appreciate any ideas.

- I have 10 tests combined from different classes, and running parallelly with CUNCURRENT execution mode. Everything looks good till now.

- Now, I have 2 tests (say for ex. 5th and 7th test) which are dependent on other tests (say 1st and 6th). So, these 2 tests should run only after the dependent tests were executed.

- **Conditional execution** didn't help as it just skips the test if condition not met, but for me it needs to execute by checking the condition again

- **ResourceLock synchronization** - This one expects a shared resource to check if any tests are sharing the same resource, and executes based on that. Not sure how this would help in this case or maybe I am missing something here.

If any ideas or approaches that we can use to achieve this or any inbuilt option we have, please share your suggestions/recommendations. Appreciate any help.

**Comment by sbrannen** [comment, marked as answer]
There is currently no support for dependencies between tests in JUnit Jupiter.

See:

- #607
- #1767
